include/deal.II/base/tensor.h

```diff
@@ -322,17 +322,17 @@ class Tensor<0, dim, Number>
    * the absolute squares of all entries. For the present case of rank-1
    * tensors, this equals the usual <tt>l<sub>2</sub></tt> norm of the vector.
    */
-  DEAL_II_CONSTEXPR real_type
-                    norm() const;
+  real_type
+  norm() const;
 
   /**
    * Return the square of the Frobenius-norm of a tensor, i.e. the sum of the
    * absolute squares of all entries.
    *
    * @note This function can also be used in CUDA device code.
    */
-  DEAL_II_CUDA_HOST_DEV real_type
-                        norm_square() const;
+  DEAL_II_CONSTEXPR DEAL_II_CUDA_HOST_DEV real_type
+                                          norm_square() const;
```

Comment:
What might be the reason that the `norm()` function can't be `constexpr`, but the `norm_square()` function can be? It seems a bit difficult at the moment to see the consistency in this `constexpr` business, and that will make it difficult to maintain all of this going forward :-(

Reply:
`std::sqrt` and `std::abs` are not required to be `constexpr`.